internal/pkg/manifest/workload.go

```diff
@@ -129,11 +129,16 @@ type Workload struct {
 
 // Image represents the workload's container image.
 type Image struct {
-	Build        BuildArgsOrString `yaml:"build"`           // Build an image from a Dockerfile.
-	Location     *string           `yaml:"location"`        // Use an existing image instead.
-	Credentials  *string           `yaml:"credentials"`     // ARN of the secret containing the private repository credentials.
-	DockerLabels map[string]string `yaml:"labels,flow"`     // Apply Docker labels to the container at runtime.
-	DependsOn    DependsOn         `yaml:"depends_on,flow"` // Add any sidecar dependencies.
+	ImageLocationOrBuild `yaml:",inline"`
+	Credentials          *string           `yaml:"credentials"`     // ARN of the secret containing the private repository credentials.
+	DockerLabels         map[string]string `yaml:"labels,flow"`     // Apply Docker labels to the container at runtime.
+	DependsOn            DependsOn         `yaml:"depends_on,flow"` // Add any sidecar dependencies.
+}
+
+// ImageLocationOrBuild represents the docker build arguments and location of the existing image.
+type ImageLocationOrBuild struct {
```

Comment:
This is generic struct that can be used by both Main container and sidecar images
I used ` Build   BuildArgsOrString` instead of `Union[*string, DockerBuildArgs]` because this will require a lot of logic and code that has to be removed such as `Unmarshal logic , validation and mergo transformers`